### PR TITLE
HHH-17310 Bug with names with digits in CamelCaseToUnderscoresNamingStrategy

### PR DESCRIPTION
**Pull request by anilabhabaral**
https://hibernate.atlassian.net/browse/HHH-17310

### CODE REVIEW

**Comment by anilabhabaral** [comment]
Hello @ChristianHoesel  @gavinking could you please take a look and review this PR. I am very new to this repository and trying to contribute in this project. Thanks.

**Comment by beikov** [comment]
Please also add a test case. Without that, we can't merge such a change.

**Comment by anilabhabaral** [comment]
Thanks @beikov I will add the test case for the changes.

**Comment by anilabhabaral** [comment]
Hi @beikov i added the testcase for the changes. Please check. Thanks!

**Comment by anilabhabaral** [comment]
Hello @beikov @mbladel could you please review this PR. Thanks!

**Comment by anilabhabaral** [comment]
Hello @beikov @mbladel could you please review this PR? Let me know if any modification is needed. Thanks

File: hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java [hunk 1]
```diff
@@ -86,7 +86,15 @@ protected boolean isCaseInsensitive(JdbcEnvironment jdbcEnvironment) {
...
 
 	private boolean isUnderscoreRequired(final char before, final char current, final char after) {
-		return Character.isLowerCase( before ) && Character.isUpperCase( current ) && Character.isLowerCase( after );
+		if (Character.isDigit(current)) {
+			if (Character.isDigit(before)) {
+				return false;
+			}
+			if (Character.isLowerCase(before)) {
+				return true;
+			}
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java:95]: So `hello1` -> `hello_1`? I don't think I see why that's correct.

It seems to me that we should simply be treating digits exactly as we treat lowercase characters, no?

**Review by gavinking — CHANGES_REQUESTED** [review]
Please see my comment.

Perhaps I've misunderstood, but then I would like to see something properly documenting all the cases that this slightly-complicated new logic is dealing with.

File: hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java [hunk 2]
```diff
@@ -86,7 +86,15 @@ protected boolean isCaseInsensitive(JdbcEnvironment jdbcEnvironment) {
...
+		}
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java:96]: A bit naively, seems to me that you could simply delete this whole block. What am I missing?

```suggestion
```

Comment (gavinking) [reply hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java:96]: Yep, sounds good to me. Thanks!

**Comment by anilabhabaral** [comment]
Hello @gavinking i have done the suggested changes. Please check.

File: hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ protected boolean isCaseInsensitive(JdbcEnvironment jdbcEnvironment) {
 	}
 
 	private boolean isUnderscoreRequired(final char before, final char current, final char after) {
-		return Character.isLowerCase( before ) && Character.isUpperCase( current ) && Character.isLowerCase( after );
+		return (Character.isLowerCase(before) || Character.isDigit(before)) && Character.isUpperCase(current) && (Character.isLowerCase(after) || Character.isDigit(after));
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/boot/model/naming/CamelCaseToUnderscoresNamingStrategy.java:89]:
```suggestion
		return ( Character.isLowerCase( before ) || Character.isDigit( before ) ) 
			&& Character.isUpperCase( current ) 
			&& ( Character.isLowerCase( after ) || Character.isDigit( after ) );
```

**Review by gavinking — CHANGES_REQUESTED** [review]
Would you please squash the commits and apply the suggested change to the code formatting?

Thanks.

**Comment by anilabhabaral** [comment]
Seems like i messed up with my git branch. I will create a new PR.

**Pull request closed without merging**